### DISCUSSION TITLE
Initialize Array Size

### DISCUSSION THREAD

**Discussion by froggestspirit** [godotengine/godot-proposals#5164] [Scripting]
I'm not sure if this was brought up in the past, but can we implement the ability to declare an array with it's size, similar to some other languages?

EX:
```
const arr_size = 10
var array[arr_size]
```

Which should function equivilently to:
```
const arr_size = 10
var array = Array().resize(arr_size)
```

**Comment by Calinou** [comment]
In general, syntax sugar is only added for things that are done very often. We want to avoid having several ways to achieve the same thing in GDScript, unless there is a good reason to.

Personally, I don't remember pre-allocating arrays like this often.
